scripts/pages/pgVideoRecorder.ts

```diff
@@ -92,70 +91,82 @@ export default class PgVideoRecorder extends PgVideoRecorderDesign {
-
-                            Multimedia.convertToMp4({
-                                videoFile: video,
-                                outputFileName: `sf-video-${new Date().toISOString().split('.')[0]}`,
-                                onCompleted: ({ video: convertedVideo }) => {
-                                    this.convertedVideo = convertedVideo;
-                                    this.lblNewSize.text = `Size with SF: ${(convertedVideo.size / (1024 * 1024)).toFixed(2)}MiB`;
-                                },
-                                onFailure: () => {
-                                    console.error('An error has occurred when video converting to mp4');
-                                }
-                            });
-                        } catch (err) {
-                            console.error('onSuccessError ', err);
-                        }
-                    },
-                    onFailure: e => {
-                        console.log(e);
-                    }
-                });
+                permissionText: global.lang.cameraPermissionFail,
+                iosPermission: permission.IOS_PERMISSIONS.CAMERA
             });
-    }
-    shareFile(video: File) {
-        if (System.OS === System.OSType.ANDROID) {
-            permission.getPermission({ androidPermission: Application.Android.Permissions.WRITE_EXTERNAL_STORAGE, permissionText: global.lang.fileAccessError })
-                .then(() => {
-                    Share.share({
-                        items: [video],
-                        page: this,
-                        blacklist: []
-                    });
-                });
-        }
-        else {
-            Share.share({
-                items: [video],
+
+            Multimedia.recordVideo({
+                ios: {
+                    cameraDevice: System.OS === System.OSType.IOS ? Multimedia.iOS.CameraDevice.FRONT : undefined
+                },
                 page: this,
-                blacklist: []
+                videoQuality: this.selectedQuality,
+                onSuccess: ({ video }) => this.recordVideoOnSuccess(video),
+                onFailure: ({ message }) => this.recordVideoOnFailure(message)
             });
+        } catch (err) {
+            console.error(err);
+        }
+    }
+    recordVideoOnSuccess(video: File) {
+        this.video = video;
+
+        this.changeRecordElementsVisible(true);
+
+        this.vwRecord.loadFile(this.video);
+
+        this.lblOldSize.text = `Size without SF: ${(this.video.size / (1024 * 1024)).toFixed(2)}MiB`;
+        this.lblNewSize.text = 'Calculating...';
+
+        Multimedia.convertToMp4({
+            videoFile: this.video,
+            outputFileName: `sf-video-${new Date().toISOString().split('.')[0]}`,
+            onCompleted: ({ video: convertedVideo }) => this.convertToMp4OnCompleted(convertedVideo),
+            onFailure: () => this.convertToMp4OnFailure
```

Comment:
Here won't work. Either do 
`onFailure: this.convertToMp4OnFailure` 
or
`onFailure: () => this.convertToMp4OnFailure()`

Reply:
I fixed it!